src/azure-cli/azure/cli/command_modules/managedservices/custom.py

```diff
@@ -0,0 +1,219 @@
+# --------------------------------------------------------------------------------------------
+import uuid
+from knack.log import get_logger
+from msrestazure.tools import parse_resource_id, is_valid_resource_id
+
+logger = get_logger(__name__)
+default_api_version = '2018-06-01-preview'
+
+
+# region Definitions Custom Commands
+
+# pylint: disable=unused-argument
+def cli_definition_create(cmd, client,
+                          name, tenant_id, principal_id, role_definition_id,
+                          plan_name=None, plan_product=None, plan_publisher=None, plan_version=None, description=None,
+                          definition_id=None, **kwargs):
+    from azure.mgmt.managedservices.models import RegistrationDefinitionProperties, Plan, \
+        Authorization
+
+    subscription = kwargs.pop('subscription', None)
+    api_version = kwargs.pop('api_version', None)
+
+    if not definition_id:
+        definition_id = str(uuid.uuid4())
+
+    sub_id = _get_subscription_id(cmd, subscription)
+    scope = _get_scope(sub_id)
+
+    if plan_name and plan_product and plan_publisher and plan_version:
+        plan = Plan(plan_name, plan_publisher, plan_product, plan_version)
+    else:
+        plan = None
+
+    authorization = Authorization(
+        principal_id=principal_id,
+        role_definition_id=role_definition_id)
+
+    authorizations = [authorization]
+
+    properties = RegistrationDefinitionProperties(
+        description=description,
+        authorizations=authorizations,
+        registration_definition_name=name,
+        managed_by_tenant_id=tenant_id,
+        Plan=plan,
+    )
+
+    return client.create_or_update(
+        registration_definition_id=definition_id,
+        api_version=_get_api_version(api_version),
+        scope=scope,
+        properties=properties)
+
+
+# pylint: disable=unused-argument
+def cli_definition_get(cmd, client,
+                       definition,
+                       **kwargs):
+    subscription = kwargs.pop('subscription', None)
+    api_version = kwargs.pop('api_version', None)
+    definition_id, sub_id, rg_name = _get_resource_id_parts(cmd, definition, subscription)
+    scope = _get_scope(sub_id, rg_name)
+    return client.get(
+        scope=scope,
+        registration_definition_id=definition_id,
+        api_version=_get_api_version(api_version))
+
+
+# pylint: disable=unused-argument
+def cli_definition_list(cmd, client, **kwargs):
+    subscription = kwargs.pop('subscription', None)
+    api_version = kwargs.pop('api_version', None)
+    sub_id = _get_subscription_id(cmd, subscription)
+    scope = _get_scope(sub_id)
+    return client.list(
+        scope=scope,
+        api_version=_get_api_version(api_version))
+
+
+def cli_definition_delete(cmd, client,
+                          definition, **kwargs):
+    subscription = kwargs.pop('subscription', None)
+    api_version = kwargs.pop('api_version', None)
```

Comment:
Why are you using kwargs?  These arguments should just be in the signature for the command.

Reply:
I was following the example in the Storage but I will change.

Reply:
Remove kwargs.